service/tls.go

```diff
@@ -17,14 +17,39 @@ limitations under the License.
 package service
 
 import (
+	"bytes"
+	"context"
+	"crypto/sha256"
 	"crypto/tls"
 	"crypto/x509"
+	"io"
 	"io/ioutil"
+	"os"
+	"sync"
+	"sync/atomic"
+	"time"
 
 	"github.com/AthenZ/authorization-proxy/v4/config"
+	"github.com/kpango/glg"
 	"github.com/pkg/errors"
 )
 
+// TLSCertificateCache represents refresh certificate
+type TLSCertificateCache struct {
```

Comment:
please change to interface when you have time, or create a new PR/issue to change it to interface.

Reply:
not fix